src/main/cpp/src/hyper_log_log_plus_plus.cu

```diff
@@ -675,7 +680,7 @@ __launch_bounds__(block_size) CUDF_KERNEL
 
   // init tmp data
   for (int i = tid; i < num_registers_per_sketch; i += block_size) {
-    shared_data[i] = 0;
+    cache[i] = 0;
```

Comment:
```suggestion
    memset(cache, 0, num_registers_per_sketch * sizeof(int));
```

Reply:
memset does not work here, each thread does ont set a continuous block of data:
```cpp
i += block_size
```